src/Model/Post/Media.php

```diff
@@ -0,0 +1,172 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Model\Post;
+
+use Friendica\Core\Logger;
+use Friendica\Core\System;
+use Friendica\Database\DBA;
+use Friendica\Util\Images;
+
+/**
+ * Class Media
+ *
+ * This Model class handles media interactions.
+ * This tables stores medias (images, videos, audio files) related to posts.
+ */
+class Media
+{
+    const UNKNOWN = 0;
+    const IMAGE   = 1;
+    const VIDEO   = 2;
+	const AUDIO   = 3;
+	const TORRENT = 16;
+
+	/**
+	 * Insert a post-media record
+	 *
+	 * @param array $media
+	 * @return void
+	 */
+	public static function insert(array $media)
+	{
+		if (empty($media['url']) || empty($media['uri-id'])) {
+			return;
+		}
+
+		if (DBA::exists('post-media', ['uri-id' => $media['uri-id'], 'url' => $media['url']])) {
+			Logger::info('Media already exists', ['uri-id' => $media['uri-id'], 'url' => $media['url'], 'callstack' => System::callstack()]);
+			return;
+		}
+
+		$fields = ['type', 'mimetype', 'height', 'width', 'size', 'preview', 'preview-height', 'preview-width', 'description'];
+		foreach ($fields as $field) {
+			if (empty($media[$field])) {
+				unset($media[$field]);
+			}
+		}
+
+		if ($media['type'] == self::IMAGE) {
+			$imagedata = Images::getInfoFromURLCached($media['url']);
+			if (!empty($imagedata)) {
+				$media['mimetype'] = $imagedata['mime'];
+				$media['size'] = $imagedata['size'];
+				$media['width'] = $imagedata[0];
+				$media['height'] = $imagedata[1];
+			}
+			if (!empty($media['preview'])) {
+				$imagedata = Images::getInfoFromURLCached($media['preview']);
+				if (!empty($imagedata)) {
+					$media['preview-width'] = $imagedata[0];
+					$media['preview-height'] = $imagedata[1];
+				}
+			}
+		}
+
+		$result = DBA::insert('post-media', $media, true);
+		Logger::info('Stored media', ['result' => $result, 'media' => $media, 'callstack' => System::callstack()]);
+	}
+
+	/**
+	 * Tests for path patterns that are usef for picture links in Friendica
+	 *
+	 * @param string $page    Link to the image page
+	 * @param string $preview Preview picture
+	 * @return boolean
+	 */
+	private static function isPictureLink(string $page, string $preview)
+	{
+		return preg_match('#/photos/.*/image/#ism', $page) && preg_match('#/photo/.*-1\.#ism', $preview);
+	}
+
+	/**
+	 * Add media links and remove them from the body
+	 *
+	 * @param integer $uriid
+	 * @param string $body
+	 * @return string Body without media links
+	 */
+	public static function addAttachmentsFromBody(int $uriid, string $body)
```

Comment:
Please rename to `insertFromBody`
```suggestion
	public static function insertFromBody(int $uriid, string $body)
```

Reply:
Done